lib/creator/yeoman/webpack-generator.js

```diff
@@ -1,138 +1,92 @@
-		this.configuration.dev.webpackOptions.resolve.unsafeCache = true;
-		this.configuration.dev.webpackOptions.resolve.plugins = [
-			"new webpack.optimize.CommonsChunkPlugin({name:" + "'" + 'vendor' + "'" + ",filename:" + "'" + 'vendor' + "-[hash].min.js'})"
-		];
-		this.configuration.dev.webpackOptions.resolve.symlinks = true;
-		this.configuration.dev.webpackOptions.resolve.cachePredicate = "function()" + "{\n return " + "true" + "\n}";
-		this.configuration.dev.webpackOptions.devtool = 'eval'
-		this.configuration.dev.webpackOptions.target = 'async-node'
-		this.configuration.dev.webpackOptions.watch = true;
-		this.configuration.dev.webpackOptions.watchOptions = {
-			aggregateTimeout: 300,
-			poll: 1000,
-			ignored: '/node_modules/'
-		}
-		this.configuration.dev.webpackOptions.externals = {
-			jquery: 'jQuery',
-			subtract: 'tracty',
-			lodash : {
-				commonjs: 'lodash',
-				amd: 'lodash',
-				root: '_' // indicates global variable
-			}
-		}
-		this.configuration.dev.webpackOptions.externals = [
-			externalRegExpFunction('^.js$')
-		]
-		this.configuration.dev.webpackOptions.externals = /^(jquery|\$)$/i
-		this.configuration.dev.webpackOptions.node = {
-			console: false,
-			global: true,
-			process: true,
-			Buffer: true,
-			__filename: 'mock',
-			__dirname: 'mock',
-			setImmediate: true
-		}
-		this.configuration.dev.webpackOptions.performance = {
-			hints: "'warning'",
-			maxEntrypointSize: 400000,
-			maxAssetSize: 100000,
-			assetFilter: assetFilterFunction('js')
-		}
-		this.configuration.dev.webpackOptions.stats = 'errors-only'
-		this.configuration.dev.webpackOptions.amd = {
-			jQuery: true,
-			kQuery: false
-		}
-		this.configuration.dev.webpackOptions.bail = true;
-		this.configuration.dev.webpackOptions.cache = 'myCache'
-		this.configuration.dev.webpackOptions.profile = true
-		this.configuration.dev.webpackOptions.module = {
-			noParse: new RegExp(/jquery|lodash/),
-			rules: [
-				{
-					test: new RegExp('/\.jsx?$/'),
-					include: [
-						/* eslint-disable */
-						'path.resolve(__dirname, "app")'
-						/* eslint-enable */
-					],
-					exclude: [
-						/* eslint-disable */
-						'path.resolve(__dirname, ".." , "app", "demo-files")'
-						/* eslint-enable */
-					],
-					enforce: "'pre'",
-					loader: "'babel-loader'",
-					options: {
-						presets: ['es2015']
-					},
-				},
-				{
-					test: new RegExp('\\.html$'),
-					use: [
-						"'htmllint-loader'",
-						{
-							loader: "'html-loader'",
-							options: {
-								hello: "'world'"
+		let done = this.async();
+		let self = this;
+
+		this.configuration.config.webpackOptions.module = getDefaultLoaders();
+		this.configuration.config.webpackOptions.plugins = getDefaultPlugins();
+		this.prompt([
+			RawList('entryType', 'What kind of entry point do you want?', ['Array', 'Function', 'String', 'Object'])
+		]).then( (entryTypeAnswer) => {
+			// Ask different questions for entry points
```

Comment:
I also wonder if we might be able to (by default) include some notion of the Webpack performance budgets to get them in that frame of mind earlier on.